tests/virt_autotest/reboot_and_wait_up_upgrade.pm

```diff
@@ -17,9 +17,32 @@ use testapi;
 use base "reboot_and_wait_up";
 
 sub run {
-    my $self    = shift;
-    my $timeout = 600;
+    my $self = shift;
+
+    #initialized to be offline upgrade
+    my $timeout = 180;
     set_var("reboot_for_upgrade_step", "yes");
+    set_var("offline_upgrade",         "yes");
+
+    #get the version that the host is installed to
+    my $host_installed_version = get_var('VERSION_TO_INSTALL', get_var('VERSION', ''));    #format 15 or 15-SP1
+    $host_installed_version =~ /^(\d+)/;
+    $host_installed_version = $1;
+    #get the version that the host should upgrade to
+    my $host_upgrade_version = get_required_var('UPGRADE_PRODUCT');                        #format sles-15-sp0
+    $host_upgrade_version =~ /sles-(\d+)-sp/;
+    $host_upgrade_version = $1;
```

Comment:
could be in one line: `($host_upgrade_version) = $host_upgrade_version =~ /sles-(\d+)-sp/;`

Reply:
Better!